src/client/Utils.ts

```diff
@@ -94,3 +94,15 @@ export const translateText = (
 
   return langSelector.translateText(key, params);
 };
+
+export class TerritoryPatternStorage {
+  private static readonly KEY = "territoryPattern";
+
+  static getSelectedPattern(): string {
+    return localStorage.getItem(TerritoryPatternStorage.KEY) || "";
```

Comment:
I think we should use `string | undefined` consistently to represent patterns.

```suggestion
  static getSelectedPattern(): string | undefined {
    return localStorage.getItem(TerritoryPatternStorage.KEY) ?? undefined;
```